components/blocks/Events.tsx

```diff
@@ -21,66 +21,32 @@ const Card = ({ cardItem, onHover }) => {
-    const eventDate = new Date(date)
-    const currentDate = new Date()
-    eventDate.setHours(0, 0, 0, 0)
-    currentDate.setHours(0, 0, 0, 0)
-
-    const timeDifference = eventDate.getTime() - currentDate.getTime()
-    const daysDifference = Math.ceil(timeDifference / (1000 * 60 * 60 * 24))
-    return daysDifference
-  }
-
-  const isDateInRange = (startDate, endDate) => {
-    const start = new Date(startDate)
-    const end = new Date(endDate)
-    const current = new Date()
-    start.setHours(0, 0, 0, 0)
-    end.setHours(0, 0, 0, 0)
-    current.setHours(0, 0, 0, 0)
-
-    return current >= start && current <= end
-  }
-
-  const formatStartDate = (date) => {
-    const d = new Date(date)
-    return `${d.getDate()}${getOrdinalSuffix(d.getDate())} ${format(d, 'MMM')}`
-  }
-
-  const formatDateRange = (start, end) => {
-    const startDate = new Date(start)
-    const endDate = new Date(end)
-    if (startDate.getMonth() === endDate.getMonth()) {
-      return `${startDate.getDate()}${getOrdinalSuffix(
-        startDate.getDate()
-      )} - ${endDate.getDate()}${getOrdinalSuffix(endDate.getDate())} ${format(
-        endDate,
-        'MMM'
-      )}`
-    }
-    return `${startDate.getDate()}${getOrdinalSuffix(
-      startDate.getDate()
-    )} ${format(startDate, 'MMM')} - ${endDate.getDate()}${getOrdinalSuffix(
-      endDate.getDate()
-    )} ${format(endDate, 'MMM')}`
+  const dateFormat = (start, end?): string => {
+    const startDay = `${new Date(start).getUTCDate() + getOrdinalSuffix(new Date(start).getUTCDate())}`
+    const startMonth = format(new Date(start), 'MMM')
+    const endDateAndHyphen = end ? ` - ${new Date(end).getUTCDate() + getOrdinalSuffix(new Date(end).getUTCDate())}` : ''
+    const endMonth = end ? format(new Date(end), 'MMM') : ''
+    return `${startDay} ${startMonth == endMonth ? '' : startMonth}${endDateAndHyphen} ${endMonth ?? startMonth}`
   }
 
   const displayDate = () => {
-    if (cardItem.startDate && cardItem.endDate) {
-      return formatDateRange(cardItem.startDate, cardItem.endDate)
-    } else if (cardItem.startDate) {
-      return formatStartDate(cardItem.startDate)
+    if (cardItem.startDate) {
+      return dateFormat(cardItem.startDate, cardItem.endDate)
     }
     return ''
   }
 
-  const daysUntilEvent = calculateDaysUntilEvent(cardItem.startDate)
-  const isPastEvent = daysUntilEvent < 0
+  const startDateUTC = new Date(Date.parse(cardItem.startDate))
+  startDateUTC.setUTCMinutes(startDateUTC.getUTCMinutes() + (cardItem.timezone * -60) + ((cardItem.startTime) * 60))
+  const endDateUTC = new Date(Date.parse(cardItem.endDate ?? cardItem.startDate))
+  endDateUTC.setUTCMinutes(endDateUTC.getUTCMinutes() + (cardItem.timezone * -60) + (24 * 60))
+  const hoursUntilEvent = Math.ceil((startDateUTC.getTime() - (new Date()).getTime()) / 36e5)
+  const hoursUntilEventEnd = Math.ceil((endDateUTC.getTime() - (new Date()).getTime()) / 36e5)
```

Comment:
this is tricky to read
Can we add some comments to help explain it?

Reply:
Done ✅